lib/iris/fileformats/um/_fast_load_structured_fields.py

```diff
@@ -99,6 +99,10 @@ def core_data(self):
         return self.data
 
     @property
+    def data_dtype(self):
+        return np.result_type(*[field.data_dtype for field in self._fields])
```

Comment:
This would seem to cause a change in behaviour from how things were before. Will this introduce the possibilty casting of the fields' data attributes that don't have the "highest" dtype?